### PR TITLE
fix problems of xprotocol goplugin example

### PR DESCRIPTION
**Pull request by nearmeng**
问题：
原有的xprotocol go plugin示例不能正常运行

解决方法：
1 codec.go 接口按照新的接口名字进行适配
2 配置中采用正确的plugin name
3 之前配置中的日志输出填写的是日志级别，已修正

### CODE REVIEW

File: examples/codes/xprotocol_with_goplugin_example/client_config.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
 										"upstream_protocol": "X",
 										"router_config_name": "client_router",
 										"extend_config": {
-											"sub_protocol": "x_example"
+											"sub_protocol": "codec"
```
Comment (doujiang24) [inline examples/codes/xprotocol_with_goplugin_example/client_config.json:41]: `x_example` is a better name for the sample.

better change the name here: https://github.com/mosn/mosn/blob/479b6bd2d03a22514e35b05588e8252ffd03d432/examples/codes/xprotocol_with_goplugin_example/codec/types.go#L26

Comment (doujiang24) [reply examples/codes/xprotocol_with_goplugin_example/client_config.json:41]: also `X` is no longer a protocol and `upstream_protocol` is useless now, better use this style:
```
"config": {
    "downstream_protocol": "x_example",
    "router_config_name": "client_router"
}
```

change details: https://mosn.io/blog/releases/v0.26.0/

**Review by doujiang24 — COMMENTED** [review]
Nice catch! Need more improvements, otherwise, LGTM.

**Review by nejisama — APPROVED** [review]
LGTM

**Review by doujiang24 — APPROVED** [review]

**Pull request merged**